ayoung/client/src/model.js

```diff
@@ -0,0 +1,156 @@
+let itemId = 1;
+
+/* NOTE: EventTarget
+    EventTarget 인터페이스는 이벤트를 수신할 수 있고,
+    수신한 이벤트에 대한 수신기를 가질 수 있는 객체가 구현하는 인터페이스.
+    즉, 이벤트의 대상이 될 수 있는 객체는 EventTarget의 세 메서드를 구현함
+    https://developer.mozilla.org/ko/docs/Web/API/EventTarget
+*/
+
+/* NOTE: 클래스나 함수를 내보낼 땐 세미콜론을 붙이지 마세요.
+    https://ko.javascript.info/import-export#ref-4122
+*/
+
+export default class TodoItems extends EventTarget {
+    constructor(todos = null) {
+        super();
+        this.items = todos || [];
+        /*
+            [
+                {
+                    id: <int, item id>,
+                    name: <str, item name>,
+                    completed: <boolean>
+                }
+            ]
+        */
+    }
+
+    create({ name }) {
+        const newItem = {
+            id: itemId,
+            name: name,
+            completed: false,
+        };
+
+        itemId++;
+
+        this.items.push(newItem);
+        this.dispatchUpdated();
+    }
+
+    get(id) {
+        const item = this.items.find(
+            (todo) => todo.id == id
+        );
+        return { ...item };
+    }
+
+    gets(filter = "All") {
+        if (filter === "All") {
+            return this.items.map(todo => ({ ...todo }));
+        }
+        if (filter === "Active") {
+            return this.items.filter(
+                (todo) => !todo.completed
+            ).map(todo => ({ ...todo }));
+        }
+        if (filter === "Completed") {
+            return this.items.filter(
+                (todo) => todo.completed
+            ).map(todo => ({ ...todo }));
+        }
+        /* 정의 안된 filter 값.. 굳이 error까진..? */
+        return this.items.map(todo => ({ ...todo }));
+    }
+
+    update(item) {
+        const target = this.items.find(
+            (todo) => todo.id == item.id
+        );
+        /*
+            TODO: 예외 처리?
+            이럴 떈 DOMError 말고 custom error를
+            정의해서 발생시키는게 적합한걸까?
+        */
+        if (target === undefined) {
+            throw "ItemNotFoundError";
+        }
```

Comment:
주로 사용되는? 또는 적합한 예외 처리 방법?

Reply:
예외처리라고 규정된 것들은 try ~ catch  혹은 위에서 처럼 throw로 예외처리가 가능합니다.

의도하신 예외처리와는 다를수 있지만, 이런 경우, falsy value를 이용해서 분기처리하여 오류를 방지할 수 있습니다. 

```suggestion
    if (!target) return
```
> target의 값은 null 아니면 특정 객체이기때문에 null 은 조건문에서 falsy value가 될 수 있어서 위에처럼 표현이 가능합니다. 

> 위에 처럼 falsy value에 해당하는 값들은 `null`, `undefined`, `''(빈문자열)`, `0`, `NaN` 가 있습니다.